PLMS.BLL/ServicesImplementation/TaskService.cs

```diff
@@ -1,206 +1,338 @@
             {
                 Title = t.Title,
                 DueDate = t.DueDate,
                 CreatedAt = t.CreatedAt
-            }).ToArray(); ;
+            }).ToArray();
 
             return results;
         }
 
-        //ToDo: apply the same as in the method above
-        public async Task<IEnumerable<TaskDto>> GetFilteredShortWithCommentsAsync(MyItemFilter filters)
+        public async Task<IEnumerable<TaskShortWithCommentsDto>> GetFilteredShortWithCommentsAsync(TaskItemFilter filters)
         {
-            IEnumerable<Task> tasks = await _taskRepository.GetFilteredAsync(
-                t => (filters.PriorityIds == null || filters.PriorityIds.Contains(t.PriorityId)) &&
-                (filters.StatusIds == null || filters.StatusIds.Contains(t.StatusId)),
-                q => q.OrderByDescending(t => t.TaskComments.Select(tc => tc.CreatedAt)),
-                "TaskComments,Goal");
-
-            if (!tasks.Any())
-            {
-                return Enumerable.Empty<TaskDto>();
-            }
+            var tasks = await GetFilteredTasksAsync(filters);
 
-            if (tasks.Count() < filters.ItemsPerPageCount)
+            if (tasks == null || !tasks.Any())
             {
-                filters.ItemsPerPageCount = tasks.Count();
+                return Enumerable.Empty<TaskShortWithCommentsDto>();
             }
 
-            var results = tasks.Take(filters.ItemsPerPageCount).Select(t =>
-            new TaskDto
+            var results = tasks.Select(t => new TaskShortWithCommentsDto
             {
                 Title = t.Title,
-                GoalTitle = t.Goal.Title,
-                Comments = t.TaskComments.Select(tс => new TaskCommentDto 
+                GoalTitle = t.Goal?.Title ?? string.Empty,
+                TaskComments = t.TaskComments?.Select(tс => new TaskCommentDto 
                 { 
                     Comment = tс.Comment, 
-                    CreatedAt = tс.CreatedAt 
-                }).ToArray()
-
+                    CreatedAt = tс.CreatedAt
+                })?
+                .ToArray() ?? Array.Empty<TaskCommentDto>(),
+                CreatedAt = t.CreatedAt
             }).ToArray();
 
             return results;
         }
 
-        public async Task<TaskDto> GetTaskByIdAsync(int id)
+        public async Task<GetTaskDto> GetTaskByIdAsync(int id, string userId)
         {
-            var task = await _taskRepository.GetByIdAsync(id);
+            var taskRepository = _unitOfWork.GetRepository<Task>();
+            var goalRepository = _unitOfWork.GetRepository<Goal>();
+            var task = await taskRepository.GetByIdAsync(id);
+            if (task == null)
+            {
+                return new GetTaskDto();
+            }
 
-            //ToDo: all null checking for entire task object as well as for some fields like Priority, Status, Goal, etc
+            var goal = await goalRepository.GetByIdAsync(task.GoalId);
+            if (goal.UserId != userId)
```

Comment:
You don't have to get goal in a separate query (and therefore you don't need goalRepository here).
You can use EF features (navigation properties & Eager / Explicit / Lazy loading) to get task.Goal.UserId, just make sure it works properly. 
Same for other entities.
Take a look at this: https://stackoverflow.com/a/75848380